### PR TITLE
feat: Adds Github Action for Update SDK

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Jira ticket: [INTMDB-1253](https://jira.mongodb.org/browse/INTMDB-1253)

Adds Github Action for Update SKD.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: cfn-resources/scripts/update-sdk-preview.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "Script will update to SDL Dev Preview. Note: you need to have the latest major version in order to be able to use it."
```
Comment (wtrocki) [inline cfn-resources/scripts/update-sdk-preview.sh:5]: Script started as helper for cli. No we have it in 3 different repos (still ok). The more we grow we should consider creating github action with those scripts or have org level workflows

**Review by wtrocki — APPROVED** [review]

File: cfn-resources/scripts/update-sdk-preview.sh (hunk 1 above)
Comment (lantoli) [reply cfn-resources/scripts/update-sdk-preview.sh:5]: yeah, an APIx common repo would be great

Comment (andreaangiolillo) [inline cfn-resources/scripts/update-sdk-preview.sh:5]:
```suggestion
echo "Script will update to SDK Dev Preview. Note: you need to have the latest major version in order to be able to use it."
```

File: .github/workflows/update-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Update SDK
```
Comment (andreaangiolillo) [inline .github/workflows/update-sdk.yml:1]: Could you add a description to this action to explain what it does?

File: cfn-resources/scripts/update-sdk-preview.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (andreaangiolillo) [inline cfn-resources/scripts/update-sdk-preview.sh:1]: when do we use this script?

File: .github/workflows/update-sdk.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+on:
+ schedule:
+   - cron: 30 8 * * TUE
```
Comment (andreaangiolillo) [inline .github/workflows/update-sdk.yml:4]:
```suggestion
   - cron: 30 8 * * TUE // Every Tuesday at 8:30 AM 
```

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**